R/write_vc.R

```diff
@@ -39,7 +40,13 @@ setMethod(
     x, file, root, sorting, override = FALSE, optimize = TRUE, ...
   ){
     if (missing(root)) {
-      stop("'root' is missing")
+      return(write_vc(x = x,
+                      file = file,
+                      root = ".",
+                      sorting = sorting,
+                      override = FALSE,
+                      optimize = TRUE,
```

Comment:
@ThierryO There is a problem here, how to fix this? While the intention is to set defaults for `override` and `optimize`, i.e. the part that says

    definition = function(
    x, file, root, sorting, override = FALSE, optimize = TRUE, ...)

, the `if (missing(root))` part, if evaluated as TRUE, returns the `write_vc` function with `override` and `optimize` set to fixed values: these values now overrule the values originally set by the user.

So, when doing something like `mydataframe %>% write_vc(file = "myfilename", sorting = mysortingvector, override = TRUE)`, while the file already existed and the sorting of the file has changed, I nevertheless get following error:

    Error: old data has different variable types or sorting, use override = TRUE

from the traceback in RStudio:

```
14.
stop(call. = FALSE, "old data has different variable types or sorting, use override = TRUE") at write_vc.R#213
13.
compare_meta(metadata, old_metadata) at write_vc.R#146
12.
write_vc(x = x, file = file, root = ".", sorting = sorting, override = FALSE, optimize = TRUE, ...) at write_vc.R#30
11.
write_vc(x = x, file = file, root = ".", sorting = sorting, override = FALSE, optimize = TRUE, ...) at write_vc.R#43
10.
write_vc(., file = "myfilename", sorting = mysortingvector, override = TRUE) at write_vc.R#30
9.
write_vc(., file = "myfilename", sorting = mysortingvector, override = TRUE)
```

Reply:
The same problem is probably present in the `rm_data()` function

Reply:
when you call a function recursively, you need to pass the arguments rather than hard coding defaults.